File: datamodel/src/main/java/datamodel/clinical/MedicationStatus.java
package datamodel.clinical;

import org.jetbrains.annotations.NotNull;

public enum MedicationStatus {
    ACTIVE("Active"),
    ON_HOLD("On hold"),
    CANCELLED("Cancelled"),
    UNKNOWN("Unknown");

    @NotNull
    private final String display;

    MedicationStatus(@NotNull final String display) {
        this.display = display;
    }

    @NotNull
    public String display() {
        return display;
    }
}

File: algo/src/main/kotlin/com/hartwig/actin/clinical/datamodel/MedicationStatus.kt
package com.hartwig.actin.clinical.datamodel

import com.hartwig.actin.Displayable

enum class MedicationStatus(private val display: String) : Displayable {
    ACTIVE("Active"),
    ON_HOLD("On hold"),
    CANCELLED("Cancelled"),
    UNKNOWN("Unknown");

    override fun display(): String {
        return display
    }
}
